src/SingleCellAnnotator.imjoy.html

```diff
@@ -627,40 +611,47 @@
           sendData.location = {};
-          sendData.location = this.current_image_data.location;
-          let noLocation = true;
-          for (loc in sendData.location){
-            if(sendData.location[loc] === null){
-              sendData.location[loc] = false
-            }
-            else{
-              noLocation = false;
-            }
-          }
-          if (noLocation && this.current_image_data.label_display_mode==='multiple'){
-            const anwser = await api.confirm("No location label selected, is that correct?")
-            if (!anwser){
-              return
+          if (this.current_task_name !== 'segmentation'){
+            sendData.location = this.current_image_data.location;
+            let noLocation = true;
+            for (loc in sendData.location){
+              if(sendData.location[loc] === null){
+                sendData.location[loc] = false
+              }
+              else{
+                noLocation = false;
+              }
             }
+            if (noLocation && this.current_image_data.label_display_mode==='multiple'){
+              const anwser = await api.confirm("No location label selected, is that correct?")
+              if (!anwser){
+                return
+              }
+            };
           };
-        }
-        
-  
+        };
         if (this.current_image_data.mask_adjusted){
           sendData.extend = this.current_image_data.extend;
           sendData.bbox = this.current_image_data.bbox;
           sendData.mask = this.current_image_data.mask_geojson;
           // now adjust the digit to 1 decimal for cooridinates
-          let coordinates = this.current_image_data.mask_geojson.features[0].geometry.coordinates[0]
-          for (let index = 0; index < coordinates.length; index++){
-            coordinates[index][0] = Number(coordinates[index][0].toFixed(1));
-            coordinates[index][1] = Number(coordinates[index][1].toFixed(1));
+          let features = this.current_image_data.mask_geojson.features
+          api.alert(JSON.stringify(features))
+          for (let i = 0; i < features.length; i++){
+            features[i].properties ={'label': 'cell'}; // keep the label as cell
+            let coordinates = features[i].geometry.coordinates[0]
+            for (let index = 0; index < coordinates.length; index++){
+              coordinates[index][0] = Number(coordinates[index][0].toFixed(1));
+              coordinates[index][1] = Number(coordinates[index][1].toFixed(1));
+            };
           };
-        }
-    
+        };
       }
       // this is the url to send the current annotation
-      const sub_url = "/annotations/" + this.current_task_name + "/" + this.current_image_data.cell_id;
+      let sub_url = "/annotations/" + this.current_task_name + "/" + this.current_image_data.cell_id;
+      if(this.current_task_name === 'segmentation'){
+        sub_url = "/annotations/" + this.current_image_data.cell_id;
```

Comment:
This is not correct, we still need the task_name.

You can remove the check here, the url is the same.

Reply:
currently, the backend is wrong at this part. It is using the format I changed

Reply:
Not for the production server right? Are you talking about the dev server? But there is no other PR for that.